src/parser/bison_parser.y

```diff
@@ -1114,6 +1120,49 @@ alias : AS IDENTIFIER { $$ = new Alias($2); }
 
 opt_alias : alias | /* empty */ { $$ = nullptr; };
 
+/******************************
+ * Row Locking Descriptions
+ ******************************/
+
+ for_locking_clause: FOR for_locking_clause_xs for_locking_clause_waiting_policy{
+   $$ = $2;
+   if($3 != 2){
+    $$->specifier = true;
+    $$->isNoWait = $3;
+   }
+ }
+ | FOR for_locking_clause_xs OF table_ref for_locking_clause_waiting_policy{
+   $$ = $2;
+   $$->depTable = $4;
+   if($5 != 2){
+    $$->specifier = true;
+    $$->isNoWait = $5;
+   }
+ }
+ | /* empty */ { $$ = nullptr; }
+
+ for_locking_clause_xs: UPDATE{
+   $$ = new ForLockingClause(true, true);
+ }
+ | NO KEY UPDATE{
+   $$ = new ForLockingClause(true, false);
+ }
+ | SHARE{
+   $$ = new ForLockingClause(false, false);
+ }
+ | KEY SHARE{
+   $$ = new ForLockingClause(false, true);
+ }
+ |/* empty */{ $$ = nullptr; };
+
+ for_locking_clause_waiting_policy: SKIP LOCKED{
+   $$ = 0;
+ }
+ | NOWAIT{
+   $$ = 1;
+ }
+ |/* empty */{ $$ = 2; };
```

Comment:
Unfortunately, integers as representatives for certain modes are not particularly descriptive. As mentioned above, I think using enumeration values would increase the readability.